test/test-runner/src/phases.rs

```diff
@@ -75,72 +75,74 @@ where
+
+            match run_result {
+                Ok(Ok(out2)) => {
+                    // There was no panic and `run` returned with a result.
+                    self.report_phases
+                        .push(PhaseReport { name: phase.name(), output: out2.test_output() });
+                    if !expect_success {
+                        return Err(PhasesError::ExpectedFailure { got: out2.test_output() });
                     }
-                    Ok(Err(err)) => {
-                        let report = tokio::runtime::Runtime::new()
-                            .unwrap()
-                            .block_on(pretty_error(&mut self.database, &self.case.uri(), err));
-                        // There was no panic and `run` returned with an error.
-                        self.report_phases
-                            .push(PhaseReport { name: phase.name(), output: report.to_string() });
-                        if expect_success {
-                            return Err(PhasesError::ExpectedSuccess { got: report });
+                    if let Some(expected) = expected_output {
+                        let actual = out2.test_output();
+                        if actual != expected {
+                            return Err(PhasesError::Mismatch {
+                                phase: phase.name(),
+                                expected,
+                                actual,
+                            });
                         }
-                        if let Some(expected) = expected_output {
-                            let actual = render_report(&report, false);
-                            if actual != expected {
-                                return Err(PhasesError::Mismatch {
-                                    phase: phase.name(),
-                                    expected,
-                                    actual,
-                                });
-                            }
-                        }
-                        Err(PhasesError::AsExpected)
                     }
-                    Err(err) => {
-                        // There was a panic
-                        self.report_phases.push(PhaseReport {
-                            name: phase.name(),
-                            output: "Panic occurred".to_string(),
-                        });
-                        Err(PhasesError::Panic { msg: err.downcast::<&str>().unwrap().to_string() })
+                    Ok(out2)
+                }
+                Ok(Err(err)) => {
+                    let reports = tokio::runtime::Runtime::new().unwrap().block_on(pretty_errors(
+                        &mut self.database,
+                        &self.case.uri(),
+                        err,
+                    ));
+
+                    // There was no panic and `run` returned with an error.
+                    self.report_phases
+                        .push(PhaseReport { name: phase.name(), output: format!("{reports:?}") });
```

Comment:
`format!("{reports:?}")` produces a debug representation of the reports. I think we might want to use `render_reports` here, as well?